compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1523,6 +1531,162 @@ toCtor env con =
+                        ETmLam (anyTpl, typeConAppToType anyTemplateTy) $
+                        ECase
+                            (EFromAny
+                                 (templateDataType tplTypeCon)
+                                 (ERecProj
+                                      anyTemplateTy
+                                      anyTemplateField
+                                      (EVar anyTpl)))
+                            [ CaseAlternative CPNone $
+                              ENone $ templateDataType tplTypeCon
+                            , CaseAlternative (CPSome tplParam) $
+                              ESome (templateDataType tplTypeCon) $
+                              EVar tplParam
+                            ]
+                    | otherwise ->
+                        EBuiltin BEError `ETyApp` lfType `ETmApp`
+                        EBuiltin
+                            (BEText
+                                 "fromAnyTemplate is not supported in this DAML-LF version")
+                "_templateTypeRep"
+                    | envLfVersion env `supports` featureTypeRep ->
+                        let resType =
+                                TypeConApp
+                                    (Qualified
+                                         stdlibRef
+                                         (mkModName ["DA", "Internal", "LF"])
+                                         (mkTypeCon ["TemplateTypeRep"]))
+                                    []
+                            resField = mkField "getTemplateTypeRep"
+                         in ERecCon
+                                resType
+                                [ ( resField
+                                  , ETypeRep $ templateDataType tplTypeCon)
+                                ]
+                    | otherwise ->
+                        EBuiltin BEError `ETyApp` lfType `ETmApp`
+                        EBuiltin
+                            (BEText
+                                 "templateTypeRep is not supported in this DAML-LF version")
+                other -> error "convertExternal: Unknown external method"
+    | otherwise = error $ "convertExternal: Unable to inline call to external method: " <> primId
+  where
+    errorExpr s =
+        EBuiltin BEError `ETyApp` lfType `ETmApp`
+        EBuiltin
+            (BEText $ "convertExternal: method " <> s <> " not implemented")
+    lookup pId modName temName = do
+        mods <- MS.lookup pId pkgIdToModules
+        mod <- NM.lookup (LF.ModuleName $ map T.pack $ splitOn "." modName) mods
+        NM.lookup (LF.TypeConName [T.pack temName]) $ LF.moduleTemplates mod
+    pkgIdToModules =
+        MS.fromList
+            [ (T.unpack $ LF.unPackageId dalfPackageId, LF.packageModules pkg)
+            | (_uId, DalfPackage {..}) <- MS.toList $ envPkgMap env
+            , let ExternalPackage _pid pkg = dalfPackagePkg
+            ]
+    anyTemplateTy =
```

Comment:
We can probably factor this out since it’s shared with the regular LF conversion.

Reply:
factored it out.